catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/RegistryPackageWebConverter.java

```diff
@@ -0,0 +1,1760 @@
+
+        if (registryMap.containsKey(REGISTRY_OBJECT_LIST_KEY)) {
+            populateRegistryObjectList((Map<String, Object>) registryMap.get(
+                    REGISTRY_OBJECT_LIST_KEY), registryPackage);
+        }
+
+        if (isRegistryPackageEmpty(registryPackage)) {
+            registryPackage = null;
+        }
+
+        return registryPackage;
+    }
+
+    private static void putTopLevel(RegistryObjectType registryObject,
+            Map<String, Object> registryMap) {
+
+        putGeneralInfo(registryObject, registryMap);
+    }
+
+    private static void populateTopLevel(Map<String, Object> registryMap,
+            RegistryObjectType registryPackage) {
+        populateGeneralInfo(registryMap, registryPackage);
+    }
+
+    private static void putRegistryPackage(RegistryPackageType registryPackage,
+            Map<String, Object> registryMap) {
+        if (registryPackage.isSetRegistryObjectList()) {
+            putRegistryObjectList(registryPackage.getRegistryObjectList(), registryMap);
+        }
+
+        if (registryPackage.isSetSlot()) {
+            putSlotList(registryPackage.getSlot(), registryMap);
+        }
+    }
+
+    private static void putRegistryObjectList(RegistryObjectListType registryObjects,
+            Map<String, Object> registryObjectMap) {
+
+        Map<String, Object> registryObjectListMap = new HashMap<>();
+
+        for (JAXBElement identifiable : registryObjects.getIdentifiable()) {
+            RegistryObjectType registryObject = (RegistryObjectType) identifiable.getValue();
+
+            putRegistryObjects(registryObject, registryObjectListMap);
+
+            if (!registryObjectListMap.isEmpty()) {
+                registryObjectMap.put(REGISTRY_OBJECT_LIST_KEY, registryObjectListMap);
+            }
+        }
+
+    }
+
+    private static void putRegistryObjects(RegistryObjectType registryObject, Map<String, Object> map) {
+        if (registryObject instanceof ExtrinsicObjectType) {
+            putExtrinsicObject((ExtrinsicObjectType) registryObject, map);
+        } else if (registryObject instanceof ServiceType) {
+            putRegistryService((ServiceType) registryObject, map);
+        } else if (registryObject instanceof OrganizationType) {
+            putRegistryOrganization((OrganizationType) registryObject, map);
+        } else if (registryObject instanceof PersonType) {
+            putRegistryPerson((PersonType) registryObject, map);
+        } else if (registryObject instanceof AssociationType1) {
+            putRegistryAssociation((AssociationType1) registryObject, map);
+        }
+    }
+
+    private static void populateRegistryObjectList(Map<String, Object> rolMap,
+            RegistryPackageType registryPackage) {
+        if (rolMap.isEmpty()) {
+            return;
+        }
+
+        RegistryObjectListType registryObjectList = RIM_FACTORY.createRegistryObjectListType();
+
+        if (rolMap.containsKey(EXTRINSIC_OBJECT_KEY)) {
```

Comment:
You _may_ be able to reduce code complexity by skipping the `containsKey()` checks in this method and replacing those with `null` checks in the called methods. We usually try to avoid passing `null`s around, but in this case it's not as bad since we're dealing with `private` methods and not a public API.